### PR TITLE
STONEBLD-683: Avoid having params in script section

### PR DESCRIPTION
**Pull request by Michkov**
Based on
https://tekton.dev/docs/pipelines/tasks/#substituting-in-script-blocks we should avoid to pass $(params.X) directly to the script section.

### CODE REVIEW

**Review by zregvart — APPROVED** [review]
Thanks :100:

**Review by dirgim — APPROVED** [review]
Looks good from my point of view!

File: task/buildah/0.1/buildah.yaml [hunk 1]
```diff
@@ -80,20 +85,20 @@ spec:
...
         cpu: 10m
     script: |
-      if [ -e "$(params.CONTEXT)/$(params.DOCKERFILE)" ]; then
-        dockerfile_path=$(params.CONTEXT)/$(params.DOCKERFILE)
-      elif [ -e "$(params.DOCKERFILE)" ]; then
-        dockerfile_path=$(params.DOCKERFILE)
-      elif echo "$(params.DOCKERFILE)" | grep -q "^https\?://"; then
-        echo "Fetch Dockerfile from $(params.DOCKERFILE)"
+      if [ -e "$CONTEXT/$DOCKERFILE" ]; then
+        dockerfile_path=$CONTEXT/$DOCKERFILE
```
Comment (MartinBasti) [inline task/buildah/0.1/buildah.yaml:89]:
```suggestion
        dockerfile_path="$CONTEXT/$DOCKERFILE"
```

File: task/buildah/0.1/buildah.yaml [hunk 2]
```diff
@@ -80,20 +85,20 @@ spec:
...
+      elif [ -e "$DOCKERFILE" ]; then
+        dockerfile_path=$DOCKERFILE
```
Comment (MartinBasti) [inline task/buildah/0.1/buildah.yaml:91]:
```suggestion
        dockerfile_path="$DOCKERFILE"
```

File: task/buildah/0.1/buildah.yaml [hunk 3]
```diff
@@ -80,20 +85,20 @@ spec:
...
+      elif echo "$DOCKERFILE" | grep -q "^https\?://"; then
+        echo "Fetch Dockerfile from $DOCKERFILE"
         dockerfile_path=$(mktemp --suffix=-Dockerfile)
-        http_code=$(curl -s -L -w "%{http_code}" --output "$dockerfile_path" $(params.DOCKERFILE))
+        http_code=$(curl -s -L -w "%{http_code}" --output "$dockerfile_path" $DOCKERFILE)
```
Comment (MartinBasti) [inline task/buildah/0.1/buildah.yaml:95]:
```suggestion
        http_code=$(curl -s -L -w "%{http_code}" --output "$dockerfile_path" "$DOCKERFILE")
```

**Review by MartinBasti — COMMENTED** [review]
In shell, variables must be quoted to prevent splitting.

I just marked few lines, but I see the whole tasks are without quoting. So probably it should be addressed globally in a separate PR

**Pull request merged**